src/Core/Cache/ApcuCacheFactory.php

```diff
@@ -27,9 +27,13 @@ public function __construct($version = null)
      */
     public function create($service, array $params = array())
     {
+        $namespace = isset($params['namespace']) ? $params['namespace'] : '';
+        $defaultLifetime = isset($params['defaultLifetime']) ? $params['defaultLifetime'] : 0;
+        $directory = isset($params['directory']) ? $params['directory'] : null;
+        $apcuNamespace = $namespace . '_' . substr(md5($directory), 0, 7);
```

Comment:
Are namespace strings length-limited? Is this substr necessary? It just risks a collision which I admit is rare but I'd only do it if it's critical.

Alternatively you could strip out bad chars from the directory.